Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.cpp

```diff
@@ -993,5 +1020,59 @@ namespace AzToolsFramework
 
             return AZStd::move(unsavedPrefabsContainer);
         }
+
+        void PrefabSaveHandler::SourceFileRemoved(
+            AZStd::string relativePath, AZStd::string scanFolder, [[maybe_unused]] AZ::Uuid sourceUUID)
+        {
+            // This gets triggered for every source file. We only need source files that are prefabs and are loaded in the current level.
+            TemplateId loadedTemplateId = s_prefabSystemComponentInterface->GetTemplateIdFromFilePath(relativePath.c_str());
+            if (loadedTemplateId != InvalidTemplateId)
+            {
+                InstanceSetConstReference instancesMappedToTemplate =
+                    m_templateInstanceMapperInterface->FindInstancesOwnedByTemplate(loadedTemplateId);
+
+                if (!instancesMappedToTemplate.has_value() || instancesMappedToTemplate->get().size() == 0)
+                {
+                    // There are no instances mapping to this template. So it can be removed from memory.
+                    s_prefabSystemComponentInterface->RemoveTemplate(loadedTemplateId);
+                }
+                else
+                {
+                    TemplateId rootTemplateId = m_prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+                    if (rootTemplateId == loadedTemplateId)
+                    {
+                        // Deletion of the current level that the user has opened needs to be handled differently. Skip for now
+                        return;
```

Comment:
question: After return, would it crash the editor?

Reply:
No it doesn't. It's the same behavior as current development. If level prefab or other prefabs within it are deleted, the loaded level stays as-is.